woocommerce/woocommerce-gutenberg-products-block
# Add block error boundary to JS-rendered blocks

Part of #1095.

This PR adds an [error boundary](https://reactjs.org/docs/error-boundaries.html) to JS-rendered blocks (_All Products_, _Reviews_ and filters), so if they throw an error, a friendly message is displayed to the user.

### Screenshots
![image](https://user-images.githubusercontent.com/3616980/68795642-a5ae4280-0651-11ea-978c-9bde6b5911dc.png)

### How to test the changes in this Pull Request:

1. Create a post with a _Reviews_ block, an _All Products_ block and a _Price Filter_ and _Attribute Filter_.
2. Add the following line [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/components/review-sort-select/index.js#L14), [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/components/product-list/index.js#L92), [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/components/price-slider/index.js#L39) and [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/components/checkbox-list/index.js#L170), so we force those blocks to throw errors:
```diff
+ throw new Error();
```
3. Preview the previous post and verify each block shows the error message.

### Changelog

> Show a friendly error message in the frontend if blocks throw a JS error.
